components/ActiveStream/index.tsx

```diff
@@ -0,0 +1,495 @@
+  const { data: nft } = useSablierV2NftDescriptorTokenUri(BigInt(tokenId));
+
+  const WithdrawAndCancel = () => {
+    return (
+      <>
+        <DialogTitle
+          sx={{
+            cursor: { xs: '', sm: 'move' },
+          }}
+        >
+          <Box
+            sx={{
+              padding: { xs: spacing(2, 1, 1), sm: spacing(1, 2, 0, 2) },
+            }}
+          >
+            <Typography fontWeight={700} fontSize={24}>
+              {t('Withdraw Reserved EXA')}
+            </Typography>
+          </Box>
+        </DialogTitle>
+        <Box
+          sx={{
+            padding: { xs: spacing(2, 1, 1), sm: spacing(2, 4, 4) },
+          }}
+        >
+          <DialogContent sx={{ p: 1, overflow: 'hidden' }}>
+            <Typography fontSize={14} fontWeight={500}>
+              {t(
+                'When you withdraw the reserved EXA, the associated vesting stream will be cancelled automatically. You’ll be able to claim the earned EXA and will get back all remaining esEXA.',
+              )}
+            </Typography>
+            <Box display="flex" gap={2} mt={4} alignItems="center">
+              {impersonateActive ? (
+                <Button fullWidth variant="contained">
+                  {t('Exit Read-Only Mode')}
+                </Button>
+              ) : chain && chain.id !== displayNetwork.id ? (
+                <LoadingButton
+                  fullWidth
+                  onClick={() => switchNetwork?.(displayNetwork.id)}
+                  variant="contained"
+                  loading={switchIsLoading}
+                >
+                  {t('Please switch to {{network}} network', { network: displayNetwork.name })}
+                </LoadingButton>
+              ) : (
+                <>
+                  <LoadingButton
+                    fullWidth
+                    variant="contained"
+                    color="error"
+                    onClick={handleCancel}
+                    loading={loading}
+                    data-testid={`vesting-stream-${tokenId}-cancel-submit`}
+                  >
+                    {t('Withdraw and Cancel Stream')}
+                  </LoadingButton>
+                </>
+              )}
+            </Box>
+          </DialogContent>
+        </Box>
+      </>
+    );
+  };
+
+  const NFT = () => {
+    const b64 = nft?.split(',')[1] ?? '';
+    const json = atob(b64) || '{}';
+    const { image, name } = JSON.parse(json);
+
+    return (
+      <Box
+        display="flex"
+        sx={{
+          padding: { xs: spacing(2, 1, 1), sm: spacing(4, 4) },
+        }}
+      >
+        <Image
+          style={{
+            borderRadius: '16px',
+          }}
+          src={image}
+          alt={name}
+          width={360}
+          height={360}
+        />
+      </Box>
+    );
+  };
+
+  const onClose = useCallback(() => {
+    setModalOpen(false);
+  }, []);
+
+  const handleContent = useCallback((content: 'nft' | 'cancel') => {
```

Comment:
why do we handle the two modals with one handler? Cant we just have a `setNFTModalOpen` and `setCancelModalOpen`?
In any case, the function doesnt 'handle' content, I'd call it openModal. Also there's other `handleClick` but's not specified which component click event is being handled.